drivers/storage/openstack/tests/openstack_test.go

```diff
@@ -0,0 +1,534 @@
+// +build !libstorage_storage_driver libstorage_storage_driver_openstack
+
+package openstack
+
+import (
+	"bytes"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	gofig "github.com/akutz/gofig/types"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/codedellemc/libstorage/api/context"
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/server"
+	apitests "github.com/codedellemc/libstorage/api/tests"
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/api/utils"
+
+	// load the driver
+	openstack "github.com/codedellemc/libstorage/drivers/storage/openstack"
+	openstackx "github.com/codedellemc/libstorage/drivers/storage/openstack/executor"
+)
+
+// configYAML is an embedded config file that specifies the bare minimum
+// configuration for creating an openstack volume.
+var configYAML = []byte(`
+openstack:
+  authURL: ` + os.Getenv("OS_AUTH_URL") + `
+  username: ` + os.Getenv("OS_USERNAME") + `
+  tenantName: ` + os.Getenv("OS_TENANT_NAME") + `
+  domainName: ` + os.Getenv("OS_USER_DOMAIN_NAME") + `
+  password: ` + os.Getenv("OS_PASSWORD") + `
+  availabilityZoneName: ` + os.Getenv("OS_AVAILABILITY_ZONE") + `
+  regionName: ` + os.Getenv("OS_REGION_NAME") + `
```

Comment:
Yes, any configuration parameters registered for the driver also get matching ENV Vars and CLI flags automatically, thanks to Viper. This is explained pretty well in our [docs](https://github.com/codedellemc/libstorage/blob/master/.docs/user-guide/config.md#configuration-properties). Since driver configs can be placed at the root of the YAML config, that means that your `openstack.domainName` config option also has a `OPENSTACK_DOMAINNAME` env var, and a `--openstackDomainName` CLI arg.

I understand wanting to use the openstack RC file env vars. Right now you have it setup such that they will be used for testing, but they are not used at all for configuring a real instance of the driver (and I think that's good). But since you are always explicitly setting the test values to those from the RC file, it means that if we run a test that use the real, normal env vars, those will be ignored.

I wonder if it is at all feasible to only try to load the RC vars from the environment if you can verify that the parameter is not already set. In short, while I am not against using the RC vars, I still want to be able to use the driver's env vars as well. Just thinking out load on that one, not even sure if it's possible.